### ISSUE TITLE
How to handle 'colliding signals' (and be successful)

### ISSUE THREAD

**Issue by ghost** [freqtrade/freqtrade#7873]
## Describe your environment

  * Operating system: Ubuntu 22.04
  * Python Version: 3.9.12
  * CCXT version: 2.2.1
  * Freqtrade Version: 2022.11.dev
  
## Your question

I want to implement a moving crossover strategy. Entry and exit trend are stated below:

Entry trend

```
dataframe.loc[
            (
                (qtpylib.crossed_above(dataframe['fast'], dataframe['slow'])) &
                (dataframe['volume'] > 0)
            ),
            'enter_long'] = 1

dataframe.loc[
            (
                (qtpylib.crossed_below(dataframe['fast'], dataframe['slow'])) &
                (dataframe['volume'] > 0)
            ),
            'enter_short'] = 1
```

Exit Trend

 ```
dataframe.loc[
            (
                (qtpylib.crossed_below(dataframe['fast'], dataframe['slow'])) &
                (dataframe['volume'] > 0)
            ),
            'exit_long'] = 1
        
dataframe.loc[
            (
                (qtpylib.crossed_above(dataframe['fast'], dataframe['slow'])) &
                (dataframe['volume'] > 0)
            ),
            'exit_short'] = 1
```

Since freqtrade does not support entry and exit on the same candle (colliding signals) I assume that a crossover signal would close the open position and open on the next candle. But instead, my position is closed and no entry is made. Rather freqtrade only make long entries (or short entries - depending what signal was made first).

I can shift(1) entry signals so that they are created after closing the previous position. But because freqtrade opens on the candle after the signal is created, the result would differ greatly.

Any suggestions how to solve this problem?

**Comment by xmatthias** [comment]
i don't think colliding signals are the point that's blocking this.
the collision would be "enter_long" / exit-short - and enter_short / exit_long - which is not a problematic collision (more info [here](https://www.freqtrade.io/en/latest/strategy-customization/#colliding-signals)).

it's still not something freqtrade supports at this point (at least not entirely).

Backtesting auto-locks the pair until the end of the candle implicitly (you can't disable this as it's built into the architecture), so you can't use backtesting for this. This automatically means you can ONLY forward-test - and therefore won't really know how the strategy will perform long-term - or in case of extreme events.

In live/dry - i guess you can get creative by unlocking the pair - but that's not something i'd recommend in general.

**Comment by ghost** [comment]
I did not mention that I am in dry-mode so your explanation makes sense. I will give it a try and forward-test my strategy, so this "issue" is done for me. Thank you for clarification!
##

Nevertheless can I add another issue in here? I dont wanna open too many.

_Exchange is Okx
1 max open trade, stake amount unlimited
trading_mode: futures
margin_mode: isolated_

Another strategy is based on heikin-ashi calculation of open and close candles. My strategy creates several entry and exit signals throughout the day.
<img width="774" alt="mystrat" src="https://user-images.githubusercontent.com/116188356/206699021-d981ca4d-73ab-4ac6-8fad-699e0416da4a.png">

Contrary to the above mentioned crossover strategy, the populate_conditions are different to each other but the entry/exit signals are created at the same candle too. Why so? 

Since freqtrade cannot open multiple positions on the same pair in parallel how would it behave? Would it enter long on the candle after the first income signal and enter short likewise (tagged blue) plus ignoring all other signals between them?

**Comment by froggleston** [comment]
Please keep issues separate. If you want to ask multiple questions, I'd recommend joining the main Freqtrade discord server, linked from the main site.

**Comment by ghost** [comment]
Will do that. Thanks.
